src/credit_default_predict_model.py

```diff
@@ -0,0 +1,157 @@
+  model_tuned : object
+      Tuned model on defined hyperparameters of Logistic regression model
+  out_dir : str
+      docopt argument --outdir
+
+  Returns
+  -------
+  None
+      Save the extracted cross validation results to random_search_cv_scores.csv in out_dir
+  """
+  final_selected_cols = ['param_logisticregression__class_weight', 'param_logisticregression__C',
+                       'split0_test_f1', 'split1_test_f1',
+                       'split2_test_f1', 'split3_test_f1',
+                       'split4_test_f1', 'split5_test_f1',
+                       'split6_test_f1', 'split7_test_f1',
+                       'split8_test_f1', 'split9_test_f1']
+  random_search_cv_scores = pd.DataFrame(model_tuned.cv_results_)[final_selected_cols]
+  random_search_cv_scores = pd.melt(random_search_cv_scores, 
+                                    id_vars=['param_logisticregression__class_weight', 'param_logisticregression__C'],
+                                    var_name='split_number',
+                                    value_name='scores')
+  # Save random_search_cv_scores to the out_dir                   
+  cv_scores_out_path = out_dir + 'random_search_cv_scores.csv'
+  try:
+    random_search_cv_scores.to_csv(cv_scores_out_path, index = False, encoding='utf-8')
+  except:
+    os.makedirs(os.path.dirname(cv_scores_out_path))
+    random_search_cv_scores.to_csv(cv_scores_out_path, index = False, encoding='utf-8')
+
+def model_hyperparameter_tuning(X_train, y_train):
+  """
+  Hyperparameter tuning on Logistic Regression model.
+
+  This function takes X_train and y_train dataframes and does model tuning using
+  RandomizedSearchCV with given set of hyperparameter C & class_weight and 
+  returns tuned model.
+
+  Parameters
+  ----------
+  X_train : dataframe
+      Input parameters in train split
+  y_train : str
+      Target column of train split
+
+  Returns
+  -------
+  object
+      Returns the tuned model object
+  """
+  scoring = [
+    "accuracy",
+    "f1",
+    "recall",
+    "precision",
+    "roc_auc",
+    "average_precision"
+  ]
+
+  param_grid = {
+    'logisticregression__C': [0.001, 0.01, 0.1, 1, 10, 100, 1000],
+    'logisticregression__class_weight':['None', 'balanced']
+  }
+  pipe = make_pipeline(LogisticRegression(max_iter=2000))
+  random_search = RandomizedSearchCV(pipe, 
+                                     param_grid, 
+                                     cv=10, 
+                                     n_jobs=-1, 
+                                     scoring=scoring, 
+                                     refit='f1', 
+                                     random_state=123)
+  random_search.fit(X_train, y_train)
+  # Test to check the return type of tuned model
+  assert isinstance(random_search, RandomizedSearchCV), "random_search is not a RandomizedSearchCV model"
+  return random_search
+
+def main(train_path, test_path, out_dir):
+  # Read train and test data from respective arguments in opt
```

Comment:
i think this file is a little over commented

Reply:
I think it is good actually